src/embeddings/config/embedding.config.ts

```
@@ -0,0 +1,126 @@
+/**
+ * Factory class for creating embedding providers
+ */
+export class EmbeddingProviderFactory {
+  /**
+   * Creates an embedding provider based on the configuration
+   * @param config The embedding system configuration
+   * @returns The created embedding provider
+   * @throws Error if the configuration is invalid
+   */
+  static createProvider(config: EmbeddingSystemConfig): IEmbeddingProvider {
+    const providerConfig = this.getProviderConfig(config);
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)